northbound/impl/src/main/java/org/opendaylight/ansible/northbound/StatusL3vpnListener.java

```diff
@@ -0,0 +1,44 @@
+package org.opendaylight.ansible.northbound;
+
+import static org.opendaylight.ansible.northbound.api.IetfL3vpnSvcUtils.STATUS_L3VPN_PATH;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.opendaylight.controller.md.sal.binding.api.DataBroker;
+import org.opendaylight.controller.md.sal.common.api.data.LogicalDatastoreType;
+import org.opendaylight.serviceutils.tools.mdsal.listener.AbstractSyncDataTreeChangeListener;
+import org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.l3vpn.svc.rev170502.StatusL3vpnProvider;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
+import org.ops4j.pax.cdi.api.OsgiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class StatusL3vpnListener extends AbstractSyncDataTreeChangeListener<StatusL3vpnProvider> {
```

Comment:
In here you are going to listen for the change where the southbound indicates it is complete with it's L3VPN sub-status right? Then once all southbound providers are complete, change overall status to complete?

Reply:
This is the flow and api we need to work out. This is the listener on the overall status and not the sub status's. There is only one provider right now so that provider can write this status for now.